modules/packages/LinearAlgebra.chpl

```diff
@@ -631,15 +631,63 @@ private proc _matmatMult(A: [?Adom] ?eltType, B: [?Bdom] eltType)
   return C;
 }
 
+pragma "no doc"
+/* Returns ``true`` if the domain is distributed */
+private proc isDistributed(a) param {
+  return !isSubtype(a.domain.dist.type, DefaultDist);
+}
 
 /* Inner product of 2 vectors. */
-proc inner(A: [?Adom], B: [?Bdom]) {
+proc inner(const ref A: [?Adom], const ref B: [?Bdom]) {
```

Comment:
What was the reason for explicit `const ref` intent here? Was it because compiler was complaining for accesses to them within `coforall`? If so, you can specify intents of the `coforall` specificially:

`coforall l in Locales with (const ref A, const ref B) do`

Reply:
No, I did it for the same reasons as declaring consts when possible in C/C++.

Reply:
We discussed this in a meeting, but just to go on record, I am fine with this. It'll make sure that we don't modify the args inside the function. Just wanted to make sure if it was done to work around some issue